system/greetd/greetd.SlackBuild

```diff
@@ -0,0 +1,218 @@
+[source.crates-io]
+registry = 'https://github.com/rust-lang/crates.io-index'
+replace-with = 'vendored-sources'
+
+[source.vendored-sources]
+directory = '$(pwd)/vendor'
+EOF
+
+# deps and versions come from Cargo.lock
+mkdir vendor
+(
+  cd vendor
+
+  grep -h -A 4 "\[\[package\]\]" \
+      $(find "../" -maxdepth 1 -mindepth 1 -name Cargo.lock | tr '\n' ' ') | \
+      sed 's/[[:space:]]*=[[:space:]]*/=/g;s/^--//;s/^\[\[/--\n[[/' | \
+      awk 'BEGIN { RS = "--\n" ; FS="\n" } { print $2, $3, $4, $5 }' | \
+      sed 's/"//g;s/name=//;s/ version=/=/' | \
+      grep " source=" | \
+      sed 's/ dependencies=.*$//' | \
+      sed 's/ source=[^ ]*//' | \
+      sed 's/ checksum=/=/' | \
+      sort -u | \
+      while read -r dep ; do
+
+    cksum="$(printf "%s\n" "$dep" | cut -d= -f3)"
+    ver="$(printf "%s\n" "$dep" | cut -d= -f2)"
+    dep="$(printf "%s\n" "$dep" | cut -d= -f1)"
+
+    tar xvf $CWD/$dep-$ver.crate
+
+    touch $dep-$ver/.cargo-ok
+
+    # generate checksum
+    {
+      printf "{\n"
+      printf '    "files": {\n'
+
+      (
+        cd $dep-$ver
+        find . -type f -print0 | xargs -0 sha256sum | sed -n '/\.cargo-checksum\.json/!p' | sed 's/\.\///;s/^\([^ ]*\)[[:space:]][[:space:]]*\(.*\)$/"\2":"\1",/'
+      ) | sed '$ s/,$//'
+
+      printf "    },\n"
+      printf '    "package": "%s"'  "$cksum"
+
+      printf "}\n"
+    } > $dep-$ver/.cargo-checksum.json
+  done
+)
+
+chown -R root:root .
+find -L . \
+ \( -perm 777 -o -perm 775 -o -perm 750 -o -perm 711 -o -perm 555 \
+  -o -perm 511 \) -exec chmod 755 {} \; -o \
+ \( -perm 666 -o -perm 664 -o -perm 640 -o -perm 600 -o -perm 444 \
+ -o -perm 440 -o -perm 400 \) -exec chmod 644 {} \;
+
+# Set /bin/bash rather than /bin/sh as default shell
+sed -i -e 's,vt = 1,vt = 7,g' config.toml
+sed -i -e 's,bin/sh,bin/bash,g' config.toml
+
+# Adding extra parms to default config to ensure our envir is good
+cat <<EOT >> config.toml
+
+[system]
+source_profile = "true"
+runfile = "/var/run/greetd.run"
+EOT
+
+## to do online rust build uncomment these two lines
+#cargo fetch --locked --target "$ARCH-unknown-linux-gnu"
+#cargo build --frozen --release --all-features
+
+## and comment the line below this
+cargo build --offline --frozen --release --all-features
+make -C man all
+
+install -Dm0755 -t "$PKG/usr/bin/" "target/release/agreety"
+install -Dm0755 -t "$PKG/usr/bin/" "target/release/fakegreet"
+install -Dm0755 -t "$PKG/usr/bin/" "target/release/$PRGNAM"
+install -Dm0644 -t "$PKG/etc/$PRGNAM/" config.toml
+install -Dm0644 "$CWD/$PRGNAM.pam" "$PKG/etc/pam.d/$PRGNAM"
+make -C man DESTDIR="$PKG" MANDIR=/usr/man PREFIX=/usr install
+
+cd $TMP/$PRGNAM-$VERSION
+
+find $PKG | xargs file | egrep "ELF.*executable" | cut -f 1 -d : \
+          | xargs strip --strip-unneeded 2> /dev/null
+
+# move this file to 'rc.4.local' for Slackware to boot directly
+# to greetd from init level 4 startup
+mkdir -p $PKG/etc/rc.d
+cp -a $CWD/rc.4.$PRGNAM $PKG/etc/rc.d/rc.4.$PRGNAM
```

Comment:
Always use `install` for things coming from `$CWD` since they are owned by whoever download the slackbuild tarball from the website or cloned the git repo and are not normalised. Or explicitly chown and chmod it

Reply:
Handled in 5a826b1c656465c5f2f9c10e2fa14f007c90524a